test/conv3d.hpp

```diff
@@ -59,4 +60,18 @@ struct conv3d_driver : conv_driver<T, ConvApi::Immediate>
     }
 };
 
-int main(int argc, const char* argv[]) { test_drive<conv3d_driver>(argc, argv); }
+class Conv3dFloat : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+
+class Conv3dHalf : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+
+class Conv3dBFloat16 : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+
+class Conv3dInt8 : public testing::TestWithParam<std::vector<std::string>>
+{
+};
```

Comment:
These definitions are used by gtests only and should reside in gtest/*.cpp files within respective namespaces.